### PR TITLE
Update Istio HPA to autoscaling/v2

### PR DESCRIPTION
**Pull request by cnvergence**
- update hpa to autoscaling/v2
- remove hpa config from cni, because it does not exist

**Related issue(s)**

See also:
- https://github.com/kyma-project/kyma/issues/16847

### CODE REVIEW

**Comment by cnvergence** [comment]
Hold for now

File: resources/istio/values.yaml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ---
 helmValues:
   global:
+    autoscalingv2API: true
```
Comment (triffer) [inline resources/istio/values.yaml:4]: We shouldn't add this as this is marked for internal usage only ([ref](https://github.com/istio/istio/pull/37872#discussion_r847543725)). This field is set depending on the configuration in the IOP.

Comment (cnvergence) [reply resources/istio/values.yaml:4]: Good point, confirmed it is working

**Review by triffer — APPROVED** [review]

**Pull request merged**